src/nvim_parinfer/core.cljs

```diff
@@ -0,0 +1,84 @@
+(ns nvim-parinfer.core
+  (:require [clojure.string :as string]
+            [parinfer :as parinfer]))
+
+(defn dbg
+  [msg & args]
+  (if (exists? js/debug)
+    (apply js/debug msg (map pr-str args))
+    (apply js/console.log msg (map pr-str args)))
+  (first args))
+
+(defn- wrap-debug-log
+  [f]
+  (fn [event]
+    (dbg "received event:" event)
+    (let [result (f event)]
+      (dbg "responding with:" result)
+      result)))
+
+(defn- vim-dict->map
+  "js->clj doesn't want to translate Vim dictionaries for us, so we force
+  the issue."
+  [x]
+  (into {} (for [k (.keys js/Object x)]
+             [k (js->clj (aget x k))])))
+
+(defn- wrap-vim-interop
```

Comment:
This was another speed issue in the past. The less `js<->cljs we do the better, it makes the processing code a bit less nice, but I think it can make a big difference: https://github.com/clojure-vim/nvim-parinfer.js/pull/14 

Feel free to benchmark on suitably sized files.